tests/prague/eip7623_increase_calldata_cost/test_transaction_validity.py

```diff
@@ -0,0 +1,524 @@
+"""  # noqa: E501
+
+from enum import Enum
+from typing import List, Sequence
+
+import pytest
+
+from ethereum_test_forks import Fork, Prague
+from ethereum_test_tools import (
+    EOA,
+    AccessList,
+    Address,
+    Alloc,
+    AuthorizationTuple,
+    Bytes,
+    Hash,
+    StateTestFiller,
+    Transaction,
+    TransactionException,
+    add_kzg_version,
+)
+
+from ...cancun.eip4844_blobs.spec import Spec as EIP_4844_Spec
+from .spec import ref_spec_7623
+
+REFERENCE_SPEC_GIT_PATH = ref_spec_7623.git_path
+REFERENCE_SPEC_VERSION = ref_spec_7623.version
+
+ENABLE_FORK = Prague
+pytestmark = [pytest.mark.valid_from(str(ENABLE_FORK))]
+
+
+class DataTestType(Enum):
+    """
+    Enum for the different types of data tests.
+    """
+
+    FLOOR_GAS_COST_LESS_THAN_OR_EQUAL_TO_INTRINSIC_GAS = 0
+    FLOOR_GAS_COST_GREATER_THAN_INTRINSIC_GAS = 1
+
+
+@pytest.fixture
+def sender(pre: Alloc) -> EOA:
+    """
+    Create the sender account.
+    """
+    return pre.fund_eoa()
+
+
+@pytest.fixture
+def to() -> Address | None:
+    """
+    Create the sender account.
+    """
+    return Address(1)
+
+
+@pytest.fixture
+def protected() -> bool:
+    """
+    Whether the transaction is protected or not. Only valid for type-0 transactions.
+    """
+    return True
+
+
+@pytest.fixture
+def access_list() -> List[AccessList] | None:
+    """
+    Access list for the transaction.
+    """
+    return None
+
+
+@pytest.fixture
+def authorization_list() -> List[AuthorizationTuple] | None:
+    """
+    Authorization list for the transaction.
+    """
+    return None
+
+
+@pytest.fixture
+def blob_versioned_hashes() -> Sequence[Hash] | None:
+    """
+    Versioned hashes for the transaction.
+    """
+    return None
+
+
+@pytest.fixture
+def contract_creating_tx(to: Address | None) -> bool:
+    """
+    Gas delta for the transaction, used to generate an invalid transaction.
+    """
+    return to is None
+
+
+@pytest.fixture
+def tx_data(
+    fork: Fork,
+    data_test_type: DataTestType,
+    access_list: List[AccessList] | None,
+    authorization_list: List[AuthorizationTuple] | None,
+    contract_creating_tx: bool,
+) -> Bytes:
+    """
+    All tests in this file use data that is generated dynamically depending on the case and the
+    attributes of the transaction in order to reach the edge cases where the floor gas cost is
+    equal or barely greater than the intrinsic gas cost.
+
+    We have two different types of tests:
+    - FLOOR_GAS_COST_LESS_THAN_OR_EQUAL_TO_INTRINSIC_GAS: The floor gas cost is less than or equal
+        to the intrinsic gas cost, which means that the size of the tokens in the data are not
+        enough to trigger the floor gas cost.
+    - FLOOR_GAS_COST_GREATER_THAN_INTRINSIC_GAS: The floor gas cost is greater than the intrinsic
+        gas cost, which means that the size of the tokens in the data are enough to trigger the
+        floor gas cost.
```

Comment:
can you also bring an example of data values that are generated here.

Reply:
Excellent, yes of course let me add an example. Thanks for the suggestion.